src/filesystem.hpp

```diff
@@ -209,18 +209,18 @@ class FileSystem {
         std::string currentWorkingDir = std::filesystem::current_path().string();
         if (givenPath.size() > 1 && givenPath.find_last_of("/\\") != std::string::npos) {
             auto configDirectory = givenPath.substr(0, givenPath.find_last_of("/\\") + 1);
-            configDirectory.empty() ? rootDirectoryPath = currentWorkingDir + "/" : rootDirectoryPath = std::move(configDirectory);
+            configDirectory.empty() ? rootDirectoryPath = currentWorkingDir + std::string(1, std::filesystem::path::preferred_separator) : rootDirectoryPath = std::move(configDirectory);
         } else {
-            rootDirectoryPath = currentWorkingDir + "/";
+            rootDirectoryPath = currentWorkingDir + std::string(1, std::filesystem::path::preferred_separator);
```

Comment:
can we make macro out of `std::string(1, std::filesystem::path::preferred_separator)`? or even declare it as constexpr so it is created only once